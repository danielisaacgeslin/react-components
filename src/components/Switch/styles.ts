import { StyleSheet } from 'aphrodite/no-important';

export default StyleSheet.create({
  container: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: 30,
    cursor: 'pointer',
    backgroundColor: '#444444',
    transition: '.4s',
    borderRadius: 34
  },
  disabled: {
    opacity: 0.5
  },
  ball: {
    backgroundColor: 'white',
    borderRadius: '100%',
    width: 22,
    minWidth: 22,
    maxWidth: 22,
    height: 22,
    minHeight: 22,
    maxHeight: 22
  },
  ballChecked: {
    flexDirection: 'row-reverse',
    right: 0,
    left: 'initial'
  },
  ballContainer: {
    padding: '0 5px',
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%'
  },
  label: {
    color: '#FFFFFF',
    fontSize: 13,
    paddingLeft: 3,
    paddingRight: 3
  }
});
