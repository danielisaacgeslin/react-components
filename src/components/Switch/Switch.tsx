import * as React from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface ISwitchProps {
  checked: boolean;
  disabled?: boolean;
  checkedColor?: string;
  checkedLabel?: string;
  unCheckedColor?: string;
  unCheckedLabel?: string;
  onChange: () => void;
}

export default class Switch extends React.PureComponent<ISwitchProps> {
  public static defaultProps = {
    disabled: false,
    checkedColor: '#444444',
    unCheckedColor: '#CCCCCC',
    checkedLabel: '',
    unCheckedLabel: ''
  };

  public onChange = () => {
    const { disabled, onChange } = this.props;
    if (!disabled) onChange();
  };

  public render() {
    const { checked, disabled, checkedColor, unCheckedColor, checkedLabel, unCheckedLabel } = this.props;
    const hasLabel = !!(checkedLabel || unCheckedLabel);
    return (
      <div key="Switch" style={{ display: 'inline-block' }} onClick={this.onChange}>
        <label
          className={css(styles.container, disabled && styles.disabled)}
          style={{ width: hasLabel ? 75 : 50, backgroundColor: checked ? checkedColor : unCheckedColor }}
        >
          <div className={css([styles.ballContainer, checked && styles.ballChecked])}>
            {hasLabel && <span className={css(styles.label)}>{checked ? checkedLabel : unCheckedLabel}</span>}
            <div className={css(styles.ball)} />
          </div>
        </label>
      </div>
    );
  }
}
