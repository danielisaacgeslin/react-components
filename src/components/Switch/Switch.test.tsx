import * as React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Switch, { ISwitchProps } from './Switch';

describe('Switch Component', () => {
  let Component;
  let props: ISwitchProps;

  beforeEach(() => {
    props = {
      checked: true,
      disabled: false,
      onChange: jest.fn()
    };
  });

  it('should render checked', () => {
    props.checked = true;
    Component = shallow(<Switch {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render unchecked', () => {
    props.checked = false;
    Component = shallow(<Switch {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render with label checked', () => {
    props.checked = true;
    props.checkedLabel = 'checked';
    props.unCheckedLabel = 'unchecked';
    Component = shallow(<Switch {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render with label unchecked', () => {
    props.checked = false;
    props.checkedLabel = 'checked';
    props.unCheckedLabel = 'unchecked';
    Component = shallow(<Switch {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render disabled', () => {
    props.disabled = true;
    Component = shallow(<Switch {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should onChange', () => {
    Component = shallow(<Switch {...props} />);
    Component.find('div')
      .first()
      .simulate('click');
    expect(props.onChange).toBeCalled();
  });

  it('should NOT onChange disabled', () => {
    props.disabled = true;
    Component = shallow(<Switch {...props} />);
    Component.find('div')
      .first()
      .simulate('click');
    expect(props.onChange).not.toBeCalled();
  });
});
