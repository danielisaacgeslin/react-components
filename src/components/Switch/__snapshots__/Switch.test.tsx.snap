// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component should render checked 1`] = `
.container {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
  height: 30px;
  cursor: pointer;
  background-color: #444444;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  transition: .4s;
  border-radius: 34px;
}

.ballContainer-ballChecked {
  -webkit-box-direction: reverse;
  -webkit-box-orient: horizontal;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-box-align: center;
  -ms-flex-align: center;
  padding: 0 5px;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  right: 0px;
  left: initial;
}

.ball {
  background-color: white;
  border-radius: 100%;
  width: 22px;
  min-width: 22px;
  max-width: 22px;
  height: 22px;
  min-height: 22px;
  max-height: 22px;
}

<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <label
    className="container"
    style={
      Object {
        "backgroundColor": "#444444",
        "width": 50,
      }
    }
  >
    <div
      className="ballContainer-ballChecked"
    >
      <div
        className="ball"
      />
    </div>
  </label>
</div>
`;

exports[`Switch Component should render disabled 1`] = `
.container-disabled {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
  height: 30px;
  cursor: pointer;
  background-color: #444444;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  transition: .4s;
  border-radius: 34px;
  opacity: 0.5;
}

.ballContainer-ballChecked {
  -webkit-box-direction: reverse;
  -webkit-box-orient: horizontal;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-box-align: center;
  -ms-flex-align: center;
  padding: 0 5px;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  right: 0px;
  left: initial;
}

.ball {
  background-color: white;
  border-radius: 100%;
  width: 22px;
  min-width: 22px;
  max-width: 22px;
  height: 22px;
  min-height: 22px;
  max-height: 22px;
}

<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <label
    className="container-disabled"
    style={
      Object {
        "backgroundColor": "#444444",
        "width": 50,
      }
    }
  >
    <div
      className="ballContainer-ballChecked"
    >
      <div
        className="ball"
      />
    </div>
  </label>
</div>
`;

exports[`Switch Component should render unchecked 1`] = `
.container {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
  height: 30px;
  cursor: pointer;
  background-color: #444444;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  transition: .4s;
  border-radius: 34px;
}

.ballContainer {
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-box-direction: normal;
  -webkit-box-orient: horizontal;
  -webkit-box-align: center;
  -ms-flex-align: center;
  padding: 0 5px;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.ball {
  background-color: white;
  border-radius: 100%;
  width: 22px;
  min-width: 22px;
  max-width: 22px;
  height: 22px;
  min-height: 22px;
  max-height: 22px;
}

<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <label
    className="container"
    style={
      Object {
        "backgroundColor": "#CCCCCC",
        "width": 50,
      }
    }
  >
    <div
      className="ballContainer"
    >
      <div
        className="ball"
      />
    </div>
  </label>
</div>
`;

exports[`Switch Component should render with label checked 1`] = `
.container {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
  height: 30px;
  cursor: pointer;
  background-color: #444444;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  transition: .4s;
  border-radius: 34px;
}

.ballContainer-ballChecked {
  -webkit-box-direction: reverse;
  -webkit-box-orient: horizontal;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-box-align: center;
  -ms-flex-align: center;
  padding: 0 5px;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  right: 0px;
  left: initial;
}

.label {
  color: #FFFFFF;
  font-size: 13px;
  padding-left: 3px;
  padding-right: 3px;
}

.ball {
  background-color: white;
  border-radius: 100%;
  width: 22px;
  min-width: 22px;
  max-width: 22px;
  height: 22px;
  min-height: 22px;
  max-height: 22px;
}

<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <label
    className="container"
    style={
      Object {
        "backgroundColor": "#444444",
        "width": 75,
      }
    }
  >
    <div
      className="ballContainer-ballChecked"
    >
      <span
        className="label"
      >
        checked
      </span>
      <div
        className="ball"
      />
    </div>
  </label>
</div>
`;

exports[`Switch Component should render with label unchecked 1`] = `
.container {
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  position: relative;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  align-items: center;
  height: 30px;
  cursor: pointer;
  background-color: #444444;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  transition: .4s;
  border-radius: 34px;
}

.ballContainer {
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  -webkit-box-direction: normal;
  -webkit-box-orient: horizontal;
  -webkit-box-align: center;
  -ms-flex-align: center;
  padding: 0 5px;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.label {
  color: #FFFFFF;
  font-size: 13px;
  padding-left: 3px;
  padding-right: 3px;
}

.ball {
  background-color: white;
  border-radius: 100%;
  width: 22px;
  min-width: 22px;
  max-width: 22px;
  height: 22px;
  min-height: 22px;
  max-height: 22px;
}

<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <label
    className="container"
    style={
      Object {
        "backgroundColor": "#CCCCCC",
        "width": 75,
      }
    }
  >
    <div
      className="ballContainer"
    >
      <span
        className="label"
      >
        unchecked
      </span>
      <div
        className="ball"
      />
    </div>
  </label>
</div>
`;
